tests/raftstore/test_stats.rs

```diff
@@ -97,3 +102,48 @@ fn test_server_simple_store_stats() {
     let mut cluster = new_server_cluster(0, 1);
     test_simple_store_stats(&mut cluster);
 }
+
+#[test]
+fn test_server_store_snap_stats() {
+    let mut cluster = new_server_cluster(0, 2);
+    cluster.cfg.store_cfg.pd_store_heartbeat_tick_interval = 20;
+
+    let pd_client = cluster.pd_client.clone();
+    // Disable default max peer number check.
+    pd_client.disable_default_rule();
+
+    let r1 = cluster.run_conf_change();
+
+    cluster.must_put(b"k1", b"v1");
+
+    // delay snapshot sending, so that we can detect this.
+    cluster.hook_transport(DelaySnapshot::new(Duration::from_millis(50)));
+    pd_client.must_add_peer(r1, new_peer(2, 2));
+
+    let snap_detected = snapshot_detected(&pd_client);
+    assert!(snap_detected, "must detect snapshot sending/receiving");
+
+    cluster.reset_transport_hooks();
+    // wait snapshot finish.
+    sleep_ms(100);
+
+    // remove the peer so we can't do any snapshot now.
+    pd_client.must_remove_peer(r1, new_peer(2, 2));
+    cluster.must_put(b"k2", b"v2");
+
+    let snap_detected = snapshot_detected(&pd_client);
```

Comment:
It will loop retry 200 times before return false. Better to pass another bool argument.